sbin/prepareWorkspace.sh

```diff
@@ -35,14 +34,55 @@ ALSA_LIB_CHECKSUM=${ALSA_LIB_CHECKSUM:-5f2cd274b272cae0d0d111e8a9e363f0878332915
 ALSA_LIB_GPGKEYID=${ALSA_LIB_GPGKEYID:-A6E59C91}
 FREETYPE_FONT_SHARED_OBJECT_FILENAME="libfreetype.so*"
 
+
+copyFromDir() {
+  echo "Copying OpenJDK source from  ${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]} to $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} to be built"
+  # we really do not want to use .git for dirs, as we expect user have them set up, ignoring them
+  local files=$(find "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}" -maxdepth 1 -mindepth 1 | grep -v -e "/workspace$" -e "/build$" -e "/.git" -e -"/build/")
+  cp -rf "$files" "./${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}/"
+}
+
+unpackFromArchive() {
+  echo "Extracting OpenJDK source tarbal ${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]} to $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} to be built"
+  # if the tarball contains .git files, they should be ignored later withut shame
+  # todo, support also zips?
+  pushd "./${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
+    local topLevelItems=$(tar --exclude='*/*' -tf  "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}" | wc -l)
+    if [ "$topLevelItems" -eq "1" ] ; then
+      echo "Source tarball contains exaclty one directory, using"
+      tar --strip-components 1 -xf "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}"
+    else
+      echo "Source tarball does not contain a top level directory, using"
+      tar --strip-components 0 -xf "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}"
+    fi
+    rm -rf "build"
+  popd
```

Comment:
```suggestion
    popd
```

Reply:
Hello! Not following this one unless you insists.   If you look carefully the block in pushd/popd is indented:
```
unpackFromArchive() {
  echo "Extracting OpenJDK source tarbal ${BUILD_CONFIG[OPENJDK_FOREST_SOURCE_ARCHIVE_ABSPATH]} to $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} to be built"
  # If the tarball contains .git files, they should be ignored later
  # todo, support also zips?
  pushd "./${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
    local topLevelItems=$(tar --exclude='*/*' -tf  "${BUILD_CONFIG[OPENJDK_FOREST_SOURCE_ARCHIVE_ABSPATH]}" | wc -l)
    if [ "$topLevelItems" -eq "1" ] ; then
      echo "Source tarball contains exaclty one directory, using"
      tar --strip-components 1 -xf "${BUILD_CONFIG[OPENJDK_FOREST_SOURCE_ARCHIVE_ABSPATH]}"
    else
      echo "Source tarball does not contain a top level directory, using"
      tar -xf "${BUILD_CONFIG[OPENJDK_FOREST_SOURCE_ARCHIVE_ABSPATH]}"
    fi
    rm -rf "build"
  popd
}
```

I don't know how global this habit is,  but its usefulness proved over years

Reply:
The indentation or popd itself? The indentation of insides of pushd/popd block is quite good habbit.